pkg/operator/sync.go

```diff
@@ -1367,6 +1372,78 @@ func (optr *Operator) isMachineOSBuilderRunning(mob *appsv1.Deployment) (bool, e
 	return false, err
 }
 
+func (optr *Operator) reconcileSimpleContentAccessSecrets(layeredMCPs []*mcfgv1.MachineConfigPool) error {
+
+	// Create set of layered and non layeredPools
+	layeredPoolSet := sets.New(layeredMCPs...)
+	allMCP, err := optr.mcpLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	allPoolSet := sets.New(allMCP...)
+	nonLayeredPoolSet := allPoolSet.Difference(layeredPoolSet)
+
+	// For all non-layered pools, if the secret exists, delete it.
+	for _, pool := range nonLayeredPoolSet.UnsortedList() {
+		secretName := ctrlcommon.SimpleContentAccessSecretName + "-" + pool.Name
+		_, err := optr.mcoSecretLister.Secrets(ctrlcommon.MCONamespace).Get(secretName)
+		if apierrors.IsNotFound(err) {
+			continue
+		} else if err != nil {
+			return err
+		}
+		if deleteErr := optr.kubeClient.CoreV1().Secrets(ctrlcommon.MCONamespace).Delete(context.TODO(), secretName, metav1.DeleteOptions{}); deleteErr != nil {
+			return deleteErr
+		}
+		klog.Infof("deleting %s", secretName)
+	}
+
+	// Now check if the Simple Content Access Secret "etc-pki-entitlement" secret exists
+	simpleContentAccessSecret, err := optr.ocManagedSecretLister.Secrets(ctrlcommon.OpenshiftConfigManagedNamespace).Get(ctrlcommon.SimpleContentAccessSecretName)
+	if apierrors.IsNotFound(err) {
+		// exit early, this cluster does not have RHEL entitlements
+		klog.V(4).Infof("Exiting early due to lack of RHEL entitlements")
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	// For all layered pools, create/update the secret
+	for _, pool := range layeredPoolSet.UnsortedList() {
+		secretName := ctrlcommon.SimpleContentAccessSecretName + "-" + pool.Name
+
+		// Create a clone of simpleContentAccessSecret, modify it to be in the MCO namespace with the new name
+		clonedSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      secretName,
+				Namespace: ctrlcommon.MCONamespace,
```

Comment:
thought (non-blocking): Something the build controller does is attach a label to each secret it creates to indicate that the secret is used for OCL. I'm not sure if attaching that label here makes sense since this isn't the build controller, but nevertheless, just thought I'd call it out.

Reply:
Good callout! Unless you feel strongly, I think I'll leave this as is (:

Reply:
I don't feel strongly about it, so that's perfectly fine!